go-dlv.el

```diff
@@ -143,27 +143,34 @@ and source-file directory for your debugger."
 (defun dlv-current-func ()
   "Debug the current program or test stopping at the beginning of the current function."
   (interactive)
-  (let (current-test-name current-func-loc)
+  (let (current-test-name current-bench-name current-func-loc)
     ;; find the location of the current function and (if it is a test function) its name
     (save-excursion
       (when (go-beginning-of-defun)
         (setq current-func-loc (format "%s:%d" buffer-file-name (line-number-at-pos)))
-        ;; if we are looking at the test function populate current-test-name
+        ;; Check for Test or Benchmark function, set current-test-name/current-bench-name
         (when (looking-at go-func-regexp)
           (let ((func-name (match-string 1)))
-            (when (and (string-match-p "_test\.go$" buffer-file-name)
-                       (string-match-p "^Test\\|^Example" func-name))
-              (setq current-test-name func-name))))))
+            (when (string-match-p "_test\.go$" buffer-file-name)
+              (cond
+               ((string-match-p "^Test\\|^Example" func-name)
+                (setq current-test-name func-name))
+               ((string-match-p "^Benchmark" func-name)
+                (setq current-bench-name func-name))))))))
 
     (if current-func-loc
         (let (gud-buffer-name dlv-command)
-          (if current-test-name
-              (progn
-                (setq gud-buffer-name "*gud-test*")
-                (setq dlv-command (concat go-dlv-command-name " test -- -test.run " current-test-name)))
-            (progn
-              (setq gud-buffer-name "*gud-debug*")
-              (setq dlv-command (concat go-dlv-command-name " debug"))))
+          (cond
+           (current-test-name
+            (setq gud-buffer-name "*gud-test*")
+            (setq dlv-command (concat go-dlv-command-name " test -- -test.run " current-test-name)))
+           (current-bench-name
+            (setq gud-buffer-name "*gud-test*")
+            (setq dlv-command (concat go-dlv-command-name " test -- -test.run BENCHMARK -test.bench " current-bench-name)))
```

Comment:
I think you can delete `-test.run BENCHMARK`, or what is the purpose of it?

Reply:
The purpose is to stop tests from running, otherwise the default will run all tests as well as the named benchmark. "BENCHMARK" can be any string that doesn't match a test name, but I can see it's not very intuitive. I'll replace it with: 
    -run=no-tests
which is more descriptive and the "-" makes it actually illegal as part of a test name so definitely won't match.

Reply:
Apologies for the huge delay!

What about `^$` for the pattern?

Reply:
Yes, ^$ is the best solution. With testing.Run() you can create tests with any string name, so the empty string is the safest non-match. testing.Run()  doesn't actually say "" is illegal as a test name, but anyone who does that deserves what they get :)